### PR TITLE
Activated Alert Rule model updates

### PR DESCRIPTION
**Pull request by nhsiehgit**
Adds relevant columns for distinguishing between `activated` alert rules vs. `continuous` alert rules

Surprisingly little is required to get this working 🤔 

We just need to add a monitor_type to the AlertRule itself in order to determine whether or not to create our subscriptions in Snuba
Once we have that, we only really need to update the SnubaQuery model to allow us to set a specified `start_time`, then we can get crackin on actually implementing the start time / dynamic query creation/deletions

### CODE REVIEW

File: src/sentry/incidents/models.py
Comment (ceorourke) [inline src/sentry/incidents/models.py:459]: doesn't this need a separate migration?

Comment (nhsiehgit) [reply src/sentry/incidents/models.py:459]: Ah yes - yes it would

I'm going to be adding all the model updates into this PR first though so we can get a wholistic view of what's going to change
then once we have signoff, i'll actually generate the migrations in separate PRs

File: src/sentry/incidents/models.py
Comment (saponifi3d) [inline src/sentry/incidents/models.py:769]:
```suggestion
    ACTIVATED = 7
    DEACTIVATED = 8
```

**Review by saponifi3d — APPROVED** [review]
LGTM!

File: src/sentry/snuba/models.py
Comment (nhsiehgit) [inline src/sentry/snuba/models.py:45]: If i want a query created to filter on a particular release, do i need to add the FK here?
Or would that just go into the `query` string above?

Comment (nhsiehgit) [reply src/sentry/snuba/models.py:45]: Actually - i suppose the `query` string is set by the user, so best not to modify that 🤔 

will keep this in place

**Comment by nhsiehgit** [comment]
Added default column value per https://develop.sentry.dev/database-migrations/#adding-columns

File: src/sentry/incidents/models.py
Comment (wedamija) [inline src/sentry/incidents/models.py:452]: Needs to be `ManyToManyField`

File: src/sentry/incidents/models.py
Comment (wedamija) [inline src/sentry/incidents/models.py:430]: I don't think we need to remove the constraint here, I'm not sure why we did it on the other table

File: src/sentry/snuba/subscriptions.py
Comment (wedamija) [inline src/sentry/snuba/subscriptions.py:36]: Is this still accurate?

File: src/sentry/snuba/subscriptions.py
Comment (wedamija) [inline src/sentry/snuba/subscriptions.py:130]: We shouldn't be passing the alert rule down to this level, this system doesn't know alert rules exist.

File: src/sentry/snuba/subscriptions.py
Comment (wedamija) [inline src/sentry/snuba/subscriptions.py:147]: Same feedback - it doesn't seem like we're getting anything out of passing alert_rule here and it's breaking the separation of concerns

**Review by wedamija — COMMENTED** [review]
Migrations/data model look good, just some other feedback too

File: src/sentry/snuba/subscriptions.py
Comment (nhsiehgit) [reply src/sentry/snuba/subscriptions.py:147]: Ah yea fair.
I initially was going to pass this down because i was going to gate the creation of the query _in_ snuba here.

but as discussed, we'll now gate the entire QuerySubscription instead

File: src/sentry/incidents/logic.py
Comment (wedamija) [inline src/sentry/incidents/logic.py:532]: It does. All the `QuerySubscription` contains is the `project_id`, and in the future the `query_extra`

File: src/sentry/incidents/models.py
Comment (wedamija) [inline src/sentry/incidents/models.py:422]: Just one extra question regarding this - Is there any chance that release monitoring alerts should actually be at the org level? So we'd just enable it for the entire org, and then we'd create subscriptions when releases are created/deployed in any project?

**Review by wedamija — COMMENTED** [review]
Migration lgtm, I'll leave it to your team for final approval, but the changes look good

File: src/sentry/incidents/models.py
Comment (nhsiehgit) [reply src/sentry/incidents/models.py:422]: I don't think so 🤔 
ActivatedAlertRules would be on a per-project basis.
We _could_ also potentially allow for full org activated rules, but that's a secondary use case atm

Comment (wedamija) [reply src/sentry/incidents/models.py:422]: Sounds good, just wanted to check!

**Review by JoshFerge — COMMENTED** [review]
nit, for especially dangerous PRs like migrations, I think it would be a good idea to try and keep changes minimal, as while the extra comments and cleanups added are helpful, it makes it slightly harder to review

**Comment by nhsiehgit** [comment]
Yea, agreed

This PR really is only comments and the migration though :/ (plus things to get all the tests happy and passing)

**Pull request merged**
